### PR TITLE
Messenger Blueprint

### PR DESCRIPTION
**Pull request by ZhouColin**
Ready for Review!

### CODE REVIEW

File: {{cookiecutter.project_slug}}/src/features/{% if cookiecutter.has_messenger_blueprint == "y" %}Messenger{% endif %}/screens/ChatScreen/index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import { GiftedChat } from 'react-native-gifted-chat'
+import PubNubReact from 'pubnub-react';
+
+export default class ChatScreen extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.pubnub = new PubNubReact({
+      publishKey: 'pub-c-09038c51-029c-47a7-95f2-b6c7407afb12',
```
Comment (indie-rok) [inline {{cookiecutter.project_slug}}/src/features/{% if cookiecutter.has_messenger_blueprint == "y" %}Messenger{% endif %}/screens/ChatScreen/index.js]: Please move this keys to the config file

https://github.com/crowdbotics/react-native-scaffold/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/src/config/app.js

File: {{cookiecutter.project_slug}}/src/features/{% if cookiecutter.has_messenger_blueprint == "y" %}Messenger{% endif %}/screens/ChatScreen/index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      subscribeKey: 'sub-c-6b1bee04-ba9f-11e9-8753-ce76e7dc5905'
+    });
+    this.pubnub.init(this);
+    state = {
```
Comment (indie-rok) [inline {{cookiecutter.project_slug}}/src/features/{% if cookiecutter.has_messenger_blueprint == "y" %}Messenger{% endif %}/screens/ChatScreen/index.js]: please use this.state while defining class property

File: {{cookiecutter.project_slug}}/src/features/{% if cookiecutter.has_messenger_blueprint == "y" %}Messenger{% endif %}/screens/ChatScreen/index.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     messages: [],
+    }
+    this.id = this.randomid();
+  }
+  randomid = () => {
+    return Math.floor(Math.random() * 100);
```
Comment (indie-rok) [inline {{cookiecutter.project_slug}}/src/features/{% if cookiecutter.has_messenger_blueprint == "y" %}Messenger{% endif %}/screens/ChatScreen/index.js]: We don't need this, you can use uuid library (added in the ptoject to generate random id's) , here are the docs:

https://github.com/kelektiv/node-uuid

File: {{cookiecutter.project_slug}}/src/features/{% if cookiecutter.has_messenger_blueprint == "y" %}Messenger{% endif %}/screens/ChatScreen/index.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  };
+
+  componentWillUnmount() {
+    this.pubnub.unsubscribe({ channels: ['ReactChat'] });
+  }
+
+  componentWillMount() {
```
Comment (indie-rok) [inline {{cookiecutter.project_slug}}/src/features/{% if cookiecutter.has_messenger_blueprint == "y" %}Messenger{% endif %}/screens/ChatScreen/index.js]: `componentWillMount` is deprecated, you might want to change this to `componentDidMount`

File: {{cookiecutter.project_slug}}/src/features/{% if cookiecutter.has_messenger_blueprint == "y" %}Messenger{% endif %}/screens/ChatScreen/index.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+  render() {
+    return (
+      <GiftedChat
+        messages={this.state.messages}
+        onSend={messages => this.onSend(messages)}
+        user={ {
+          _id: this.id,
+          name: 'React Native',
+          avatar: 'https://cdn-images-1.medium.com/max/1200/1*D0E03PmeVga1wLJt7YLd1w.png',
```
Comment (indie-rok) [inline {{cookiecutter.project_slug}}/src/features/{% if cookiecutter.has_messenger_blueprint == "y" %}Messenger{% endif %}/screens/ChatScreen/index.js]: I think we have an avatar image in the fixed folder, you can import it using `require('../../assets/image-url.png')`

**Comment by tan-duong** [comment]
using the UUID generator is done

**Pull request merged**